### ISSUE TITLE
Unable to load MISP modules from package.

### ISSUE THREAD

**Issue by maisenhe** [MISP/misp-modules#126]
After updating to the newest MISP-modules package today. I ran the following commands

cd misp-modules
sudo pip3 install -I -r REQUIREMENTS
sudo pip3 install -I .
sudo vi /etc/rc.local, add this line: `sudo -u www-data misp-modules -s &`

after this I get the following error

2017-06-06 10:36:17,125 - misp-modules - INFO - Unable to load MISP modules from package.

before I did this update the modules were working just fine.

**Comment by Rafiot** [comment]
This is weird.

Can you paste the output of the other commands?

**Comment by maisenhe** [comment]
Sure here is the output from the build. I even went as far to spin up another VM and build MISP and the modules from scratch and I got the same results. Also here is the type of machine I’m running this on . Thanks for your help in advance.

lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty



root@misp:/usr/local/src# sudo git clone https://github.com/MISP/misp-modules.git
Cloning into 'misp-modules'...
remote: Counting objects: 1696, done.
remote: Total 1696 (delta 0), reused 0 (delta 0), pack-reused 1696
Receiving objects: 100% (1696/1696), 300.69 KiB | 0 bytes/s, done.
Resolving deltas: 100% (1040/1040), done.
root@misp:/usr/local/src# cd misp-modules
root@misp:/usr/local/src/misp-modules# sudo pip3 install -I -r REQUIREMENTS
Collecting stix (from -r REQUIREMENTS (line 1))
  Downloading stix-1.2.0.4-py2.py3-none-any.whl (289kB)
    100% |████████████████████████████████| 296kB 906kB/s
Collecting cybox (from -r REQUIREMENTS (line 2))
  Downloading cybox-2.1.0.14-py2.py3-none-any.whl (724kB)
    100% |████████████████████████████████| 727kB 529kB/s
Collecting tornado (from -r REQUIREMENTS (line 3))
  Downloading tornado-4.5.1.tar.gz (483kB)
    100% |████████████████████████████████| 491kB 630kB/s
Collecting dnspython3 (from -r REQUIREMENTS (line 4))
  Downloading dnspython3-1.15.0.zip
Collecting requests (from -r REQUIREMENTS (line 5))
  Downloading requests-2.17.3-py2.py3-none-any.whl (87kB)
    100% |████████████████████████████████| 92kB 1.4MB/s
Collecting urlarchiver (from -r REQUIREMENTS (line 6))
  Downloading urlarchiver-0.2.tar.gz
Collecting passivetotal (from -r REQUIREMENTS (line 7))
  Downloading passivetotal-1.0.30.tar.gz
Collecting PyPDNS (from -r REQUIREMENTS (line 8))
  Downloading pypdns-1.3.tar.gz
Collecting pypssl (from -r REQUIREMENTS (line 9))
  Downloading pypssl-2.1.tar.gz
Collecting redis (from -r REQUIREMENTS (line 10))
  Downloading redis-2.10.5-py2.py3-none-any.whl (60kB)
    100% |████████████████████████████████| 61kB 1.3MB/s
Collecting pyeupi (from -r REQUIREMENTS (line 11))
  Downloading pyeupi-1.0.tar.gz
Collecting ipasn-redis (from -r REQUIREMENTS (line 12))
  Downloading ipasn-redis-2.0.tar.gz
Collecting asnhistory (from -r REQUIREMENTS (line 13))
  Downloading asnhistory-2.0.4.tar.gz
Collecting uwhois from git+https://github.com/Rafiot/uwhoisd.git@testing#egg=uwhois&subdirectory=client (from -r REQUIREMENTS (line 14))
  Cloning https://github.com/Rafiot/uwhoisd.git (to testing) to /tmp/pip-build-uwg0z__v/uwhois
Collecting misp_stix_converter from git+https://github.com/MISP/MISP-STIX-Converter.git#egg=misp_stix_converter (from -r REQUIREMENTS (line 15))
  Cloning https://github.com/MISP/MISP-STIX-Converter.git to /tmp/pip-build-uwg0z__v/misp-stix-converter
Collecting pymisp from git+https://github.com/MISP/PyMISP.git#egg=pymisp (from -r REQUIREMENTS (line 16))
  Cloning https://github.com/MISP/PyMISP.git to /tmp/pip-build-uwg0z__v/pymisp
Collecting pillow (from -r REQUIREMENTS (line 17))
  Downloading Pillow-4.1.1-cp35-cp35m-manylinux1_x86_64.whl (5.7MB)
    100% |████████████████████████████████| 5.7MB 89kB/s
Collecting pytesseract (from -r REQUIREMENTS (line 18))
  Downloading pytesseract-0.1.7.tar.gz (150kB)
    100% |████████████████████████████████| 153kB 1.1MB/s
Collecting SPARQLWrapper (from -r REQUIREMENTS (line 19))
  Downloading SPARQLWrapper-1.8.0.zip
Collecting domaintools_api (from -r REQUIREMENTS (line 20))
  Downloading domaintools_api-0.1.7.tar.gz
Collecting pygeoip (from -r REQUIREMENTS (line 21))
  Downloading pygeoip-0.3.2-py2.py3-none-any.whl
Collecting bs4 (from -r REQUIREMENTS (line 22))
  Downloading bs4-0.0.1.tar.gz
Collecting lxml>=2.3 (from stix->-r REQUIREMENTS (line 1))
  Downloading lxml-3.8.0-cp35-cp35m-manylinux1_x86_64.whl (7.2MB)
    100% |████████████████████████████████| 7.2MB 90kB/s
Collecting python-dateutil (from stix->-r REQUIREMENTS (line 1))
  Downloading python_dateutil-2.6.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 962kB/s
Collecting mixbox>=1.0.2 (from stix->-r REQUIREMENTS (line 1))
  Downloading mixbox-1.0.2-py2.py3-none-any.whl (46kB)
    100% |████████████████████████████████| 51kB 1.7MB/s
Collecting dnspython==1.15.0 (from dnspython3->-r REQUIREMENTS (line 4))
  Downloading dnspython-1.15.0-py2.py3-none-any.whl (177kB)
    100% |████████████████████████████████| 184kB 1.1MB/s
Collecting chardet<3.1.0,>=3.0.2 (from requests->-r REQUIREMENTS (line 5))
  Downloading chardet-3.0.3-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 895kB/s
Collecting certifi>=2017.4.17 (from requests->-r REQUIREMENTS (line 5))
  Downloading certifi-2017.4.17-py2.py3-none-any.whl (375kB)
    100% |████████████████████████████████| 378kB 759kB/s
Collecting urllib3<1.22,>=1.21.1 (from requests->-r REQUIREMENTS (line 5))
  Downloading urllib3-1.21.1-py2.py3-none-any.whl (131kB)
    100% |████████████████████████████████| 133kB 758kB/s
Collecting idna<2.6,>=2.5 (from requests->-r REQUIREMENTS (line 5))
  Downloading idna-2.5-py2.py3-none-any.whl (55kB)
    100% |████████████████████████████████| 61kB 946kB/s
Collecting url-normalize (from urlarchiver->-r REQUIREMENTS (line 6))
  Downloading url_normalize-1.3.1-py3-none-any.whl
Collecting ez_setup (from passivetotal->-r REQUIREMENTS (line 7))
  Downloading ez_setup-0.9.tar.gz
Collecting future (from passivetotal->-r REQUIREMENTS (line 7))
  Downloading future-0.16.0.tar.gz (824kB)
    100% |████████████████████████████████| 829kB 479kB/s
Collecting requests-cache (from PyPDNS->-r REQUIREMENTS (line 8))
  Downloading requests_cache-0.4.13-py2.py3-none-any.whl
Collecting dateutils (from asnhistory->-r REQUIREMENTS (line 13))
  Downloading dateutils-0.6.6.tar.gz
Collecting cabby (from misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading cabby-0.1.17.tar.gz (362kB)
    100% |████████████████████████████████| 368kB 529kB/s
Collecting nose (from misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading nose-1.3.7-py3-none-any.whl (154kB)
    100% |████████████████████████████████| 163kB 998kB/s
Collecting pyaml>=3.11 (from misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading pyaml-16.12.2-py2.py3-none-any.whl
Collecting jsonschema (from pymisp->-r REQUIREMENTS (line 16))
  Downloading jsonschema-2.6.0-py2.py3-none-any.whl
Collecting olefile (from pillow->-r REQUIREMENTS (line 17))
  Downloading olefile-0.44.zip (74kB)
    100% |████████████████████████████████| 81kB 1.2MB/s
Collecting rdflib>=4.0 (from SPARQLWrapper->-r REQUIREMENTS (line 19))
  Downloading rdflib-4.2.2-py3-none-any.whl (344kB)
    100% |████████████████████████████████| 348kB 835kB/s
Collecting aiohttp==1.0.5 (from domaintools_api->-r REQUIREMENTS (line 20))
  Downloading aiohttp-1.0.5-cp35-cp35m-manylinux1_x86_64.whl (150kB)
    100% |████████████████████████████████| 153kB 1.3MB/s
Collecting beautifulsoup4 (from bs4->-r REQUIREMENTS (line 22))
  Downloading beautifulsoup4-4.6.0-py3-none-any.whl (86kB)
    100% |████████████████████████████████| 92kB 1.5MB/s
Collecting six>=1.5 (from python-dateutil->stix->-r REQUIREMENTS (line 1))
  Downloading six-1.10.0-py2.py3-none-any.whl
Collecting ordered-set (from mixbox>=1.0.2->stix->-r REQUIREMENTS (line 1))
  Downloading ordered-set-2.0.2.tar.gz
Collecting argparse (from dateutils->asnhistory->-r REQUIREMENTS (line 13))
  Downloading argparse-1.4.0-py2.py3-none-any.whl
Collecting pytz (from dateutils->asnhistory->-r REQUIREMENTS (line 13))
  Downloading pytz-2017.2-py2.py3-none-any.whl (484kB)
    100% |████████████████████████████████| 491kB 601kB/s
Collecting colorlog (from cabby->misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading colorlog-2.10.0-py2.py3-none-any.whl
Collecting furl>=0.4.7 (from cabby->misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading furl-1.0.0.tar.gz
Collecting libtaxii>=1.1.109 (from cabby->misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading libtaxii-1.1.110.zip (143kB)
    100% |████████████████████████████████| 153kB 1.0MB/s
Collecting PyYAML (from pyaml>=3.11->misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading PyYAML-3.12.tar.gz (253kB)
    100% |████████████████████████████████| 256kB 468kB/s
Collecting isodate (from rdflib>=4.0->SPARQLWrapper->-r REQUIREMENTS (line 19))
  Downloading isodate-0.5.4.tar.gz
Collecting pyparsing (from rdflib>=4.0->SPARQLWrapper->-r REQUIREMENTS (line 19))
  Downloading pyparsing-2.2.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 1.2MB/s
Collecting multidict>=2.0 (from aiohttp==1.0.5->domaintools_api->-r REQUIREMENTS (line 20))
  Downloading multidict-2.1.6-cp35-cp35m-manylinux1_x86_64.whl (357kB)
    100% |████████████████████████████████| 358kB 617kB/s
Collecting async-timeout (from aiohttp==1.0.5->domaintools_api->-r REQUIREMENTS (line 20))
  Downloading async_timeout-1.2.1-py3-none-any.whl
Collecting orderedmultidict>=0.7.8 (from furl>=0.4.7->cabby->misp_stix_converter->-r REQUIREMENTS (line 15))
  Downloading orderedmultidict-0.7.11.tar.gz
Building wheels for collected packages: tornado, dnspython3, urlarchiver, passivetotal, PyPDNS, pypssl, pyeupi, ipasn-redis, asnhistory, pytesseract, SPARQLWrapper, domaintools-api, bs4, ez-setup, future, dateutils, cabby, olefile, ordered-set, furl, libtaxii, PyYAML, isodate, orderedmultidict
  Running setup.py bdist_wheel for tornado ... done
  Stored in directory: /root/.cache/pip/wheels/84/83/cd/6a04602633457269d161344755e6766d24307189b7a67ff4b7
  Running setup.py bdist_wheel for dnspython3 ... done
  Stored in directory: /root/.cache/pip/wheels/f2/58/6e/265f4a6753a691c71717637031a9ddfcf47162ea02335ceb48
  Running setup.py bdist_wheel for urlarchiver ... done
  Stored in directory: /root/.cache/pip/wheels/16/7d/b1/4599539932935906d352a30e6a80e7f9939ee3d1f0bf362df3
  Running setup.py bdist_wheel for passivetotal ... done
  Stored in directory: /root/.cache/pip/wheels/0a/19/fd/1f0f52285cc241ea5457e5ed0d62e989ca7f38036df53e7c88
  Running setup.py bdist_wheel for PyPDNS ... done
  Stored in directory: /root/.cache/pip/wheels/ea/25/43/85d4874262333efbab38e66db6551b3e1c82ae5095185446ec
  Running setup.py bdist_wheel for pypssl ... done
  Stored in directory: /root/.cache/pip/wheels/4e/ec/48/e17476bcce9ea0830424a99f8f931f15a8d19ed0204a1af7cd
  Running setup.py bdist_wheel for pyeupi ... done
  Stored in directory: /root/.cache/pip/wheels/1c/a5/94/fc51085fa41b5f3dc3ecbd8123ed9057e4bfb37e63bf8c4f79
  Running setup.py bdist_wheel for ipasn-redis ... done
  Stored in directory: /root/.cache/pip/wheels/dc/f5/5b/b39a883ba5d36722909164e3a2c20bedd67dd948fbaf9a7cfd
  Running setup.py bdist_wheel for asnhistory ... done
  Stored in directory: /root/.cache/pip/wheels/aa/44/4d/ce895e2eaa154fda3031398d2439cdad10a9cb249d38a9efa3
  Running setup.py bdist_wheel for pytesseract ... done
  Stored in directory: /root/.cache/pip/wheels/ff/06/21/4b0d1e4d3e0e7a1c46c72a8a66a629473316f8e02614e9fdda
  Running setup.py bdist_wheel for SPARQLWrapper ... done
  Stored in directory: /root/.cache/pip/wheels/3c/b6/36/90a0877c0a7ac2933a16c446b535af632831b3fd692e1910fe
  Running setup.py bdist_wheel for domaintools-api ... done
  Stored in directory: /root/.cache/pip/wheels/d3/b8/16/ab6be0c0e3f4cb57ef4f5431814c4e32264c08c90651d5a960
  Running setup.py bdist_wheel for bs4 ... done
  Stored in directory: /root/.cache/pip/wheels/84/67/d4/9e09d9d5adede2ee1c7b7e8775ba3fbb04d07c4f946f0e4f11
  Running setup.py bdist_wheel for ez-setup ... done
  Stored in directory: /root/.cache/pip/wheels/53/d6/8b/a2270e66cf5f68c038fffcc8b871de0c4326550fb187906bab
  Running setup.py bdist_wheel for future ... done
  Stored in directory: /root/.cache/pip/wheels/c2/50/7c/0d83b4baac4f63ff7a765bd16390d2ab43c93587fac9d6017a
  Running setup.py bdist_wheel for dateutils ... done
  Stored in directory: /root/.cache/pip/wheels/61/60/af/2081d0cb3cfaebaab704b6012d5b1a3bdba9e0c4be33ff7e65
  Running setup.py bdist_wheel for cabby ... done
  Stored in directory: /root/.cache/pip/wheels/1e/cc/cb/98f724e29f8f0ec0166598374a9b10c53244c1658182fffd22
  Running setup.py bdist_wheel for olefile ... done
  Stored in directory: /root/.cache/pip/wheels/20/58/49/cc7bd00345397059149a10b0259ef38b867935ea2ecff99a9b
  Running setup.py bdist_wheel for ordered-set ... done
  Stored in directory: /root/.cache/pip/wheels/f4/43/ca/fc4c4db9a49c68da81266de2718896dc6a57e7a309e3a80d96
  Running setup.py bdist_wheel for furl ... done
  Stored in directory: /root/.cache/pip/wheels/cd/b5/aa/7dc5e7065463108ae8909455822096ee1618c16cc972117d31
  Running setup.py bdist_wheel for libtaxii ... done
  Stored in directory: /root/.cache/pip/wheels/9a/18/5c/c1793da503662d644a93ca0fa593188ded576d21ed7dede38f
  Running setup.py bdist_wheel for PyYAML ... done
  Stored in directory: /root/.cache/pip/wheels/2c/f7/79/13f3a12cd723892437c0cfbde1230ab4d82947ff7b3839a4fc
  Running setup.py bdist_wheel for isodate ... done
  Stored in directory: /root/.cache/pip/wheels/61/c0/d2/6b4a10c222ba9261ab9872a8f05d471652962284e8c677e5e7
  Running setup.py bdist_wheel for orderedmultidict ... done
  Stored in directory: /root/.cache/pip/wheels/71/1a/2f/4de152f04d129bd5f4ac805a75f944be3b1e4af4d17a18ed54
Successfully built tornado dnspython3 urlarchiver passivetotal PyPDNS pypssl pyeupi ipasn-redis asnhistory pytesseract SPARQLWrapper domaintools-api bs4 ez-setup future dateutils cabby olefile ordered-set furl libtaxii PyYAML isodate orderedmultidict
Installing collected packages: lxml, six, python-dateutil, ordered-set, mixbox, cybox, stix, tornado, dnspython, dnspython3, chardet, certifi, urllib3, idna, requests, url-normalize, urlarchiver, ez-setup, future, passivetotal, requests-cache, PyPDNS, pypssl, redis, pyeupi, ipasn-redis, argparse, pytz, dateutils, asnhistory, uwhois, colorlog, orderedmultidict, furl, libtaxii, cabby, nose, PyYAML, pyaml, jsonschema, pymisp, misp-stix-converter, olefile, pillow, pytesseract, isodate, pyparsing, rdflib, SPARQLWrapper, multidict, async-timeout, aiohttp, domaintools-api, pygeoip, beautifulsoup4, bs4
  Running setup.py install for uwhois ... done
  Running setup.py install for pymisp ... done
  Running setup.py install for misp-stix-converter ... done
Successfully installed PyPDNS-1.3 PyYAML-3.12 SPARQLWrapper-1.8.0 aiohttp-1.0.5 argparse-1.4.0 asnhistory-2.0.4 async-timeout-1.2.1 beautifulsoup4-4.6.0 bs4-0.0.1 cabby-0.1.17 certifi-2017.4.17 chardet-3.0.3 colorlog-2.10.0 cybox-2.1.0.14 dateutils-0.6.6 dnspython-1.15.0 dnspython3-1.15.0 domaintools-api-0.1.7 ez-setup-0.9 furl-1.0.0 future-0.16.0 idna-2.5 ipasn-redis-2.0 isodate-0.5.4 jsonschema-2.6.0 libtaxii-1.1.110 lxml-3.8.0 misp-stix-converter-0.2.9 mixbox-1.0.2 multidict-2.1.6 nose-1.3.7 olefile-0.44 ordered-set-2.0.2 orderedmultidict-0.7.11 passivetotal-1.0.30 pillow-4.1.1 pyaml-16.12.2 pyeupi-1.0 pygeoip-0.3.2 pymisp-2.4.71 pyparsing-2.2.0 pypssl-2.1 pytesseract-0.1.7 python-dateutil-2.6.0 pytz-2017.2 rdflib-4.2.2 redis-2.10.5 requests-2.17.3 requests-cache-0.4.13 six-1.10.0 stix-1.2.0.4 tornado-4.5.1 url-normalize-1.3.1 urlarchiver-0.2 urllib3-1.21.1 uwhois-0.5
root@misp:/usr/local/src/misp-modules# sudo pip3 install -I .
Processing /usr/local/src/misp-modules
Collecting PyPDNS (from misp-modules==1.0)
Collecting asnhistory (from misp-modules==1.0)
Collecting cybox (from misp-modules==1.0)
  Using cached cybox-2.1.0.14-py2.py3-none-any.whl
Collecting dnspython3 (from misp-modules==1.0)
Collecting ipasn-redis (from misp-modules==1.0)
Collecting passivetotal (from misp-modules==1.0)
Collecting pillow (from misp-modules==1.0)
  Using cached Pillow-4.1.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting pyeupi (from misp-modules==1.0)
Collecting pypssl (from misp-modules==1.0)
Collecting pytesseract (from misp-modules==1.0)
Collecting redis (from misp-modules==1.0)
  Using cached redis-2.10.5-py2.py3-none-any.whl
Collecting requests (from misp-modules==1.0)
  Using cached requests-2.17.3-py2.py3-none-any.whl
Collecting shodan (from misp-modules==1.0)
  Downloading shodan-1.6.8.tar.gz
Collecting stix (from misp-modules==1.0)
  Using cached stix-1.2.0.4-py2.py3-none-any.whl
Collecting tornado (from misp-modules==1.0)
Collecting urlarchiver (from misp-modules==1.0)
Collecting requests-cache (from PyPDNS->misp-modules==1.0)
  Using cached requests_cache-0.4.13-py2.py3-none-any.whl
Collecting dateutils (from asnhistory->misp-modules==1.0)
Collecting mixbox>=1.0.2 (from cybox->misp-modules==1.0)
  Using cached mixbox-1.0.2-py2.py3-none-any.whl
Collecting lxml>=2.2.3 (from cybox->misp-modules==1.0)
  Using cached lxml-3.8.0-cp35-cp35m-manylinux1_x86_64.whl
Collecting python-dateutil (from cybox->misp-modules==1.0)
  Using cached python_dateutil-2.6.0-py2.py3-none-any.whl
Collecting dnspython==1.15.0 (from dnspython3->misp-modules==1.0)
  Using cached dnspython-1.15.0-py2.py3-none-any.whl
Collecting ez-setup (from passivetotal->misp-modules==1.0)
Collecting future (from passivetotal->misp-modules==1.0)
Collecting olefile (from pillow->misp-modules==1.0)
Collecting idna<2.6,>=2.5 (from requests->misp-modules==1.0)
  Using cached idna-2.5-py2.py3-none-any.whl
sudo -u www-data bash /var/www/MISP/app/Console/worker/start.sh
Collecting urllib3<1.22,>=1.21.1 (from requests->misp-modules==1.0)
  Using cached urllib3-1.21.1-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests->misp-modules==1.0)
  Using cached certifi-2017.4.17-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests->misp-modules==1.0)
  Using cached chardet-3.0.3-py2.py3-none-any.whl
Collecting click (from shodan->misp-modules==1.0)
  Downloading click-6.7-py2.py3-none-any.whl (71kB)
    100% |████████████████████████████████| 71kB 649kB/s
Collecting click-plugins (from shodan->misp-modules==1.0)
  Downloading click-plugins-1.0.3.tar.gz
Collecting colorama (from shodan->misp-modules==1.0)
  Downloading colorama-0.3.9-py2.py3-none-any.whl
Collecting url-normalize (from urlarchiver->misp-modules==1.0)
  Using cached url_normalize-1.3.1-py3-none-any.whl
Collecting pytz (from dateutils->asnhistory->misp-modules==1.0)
  Using cached pytz-2017.2-py2.py3-none-any.whl
Collecting argparse (from dateutils->asnhistory->misp-modules==1.0)
  Using cached argparse-1.4.0-py2.py3-none-any.whl
Collecting ordered-set (from mixbox>=1.0.2->cybox->misp-modules==1.0)
Collecting six>=1.5 (from python-dateutil->cybox->misp-modules==1.0)
  Using cached six-1.10.0-py2.py3-none-any.whl
Building wheels for collected packages: misp-modules, shodan, click-plugins
  Running setup.py bdist_wheel for misp-modules ... done
  Stored in directory: /root/.cache/pip/wheels/84/23/c5/f741153ef3c925d089a7d56c8b45251e71b4a058385cbabec3
  Running setup.py bdist_wheel for shodan ... done
  Stored in directory: /root/.cache/pip/wheels/de/a5/9d/29bd7ac0e8e69c951974ef4e98a99892c56eed427fe386cc03
  Running setup.py bdist_wheel for click-plugins ... done
  Stored in directory: /root/.cache/pip/wheels/70/d7/3d/188128669f2aa42f6008217d9e2e6826d398dc3e361c0fcb75
Successfully built misp-modules shodan click-plugins
Installing collected packages: idna, urllib3, certifi, chardet, requests, requests-cache, PyPDNS, redis, pytz, argparse, six, python-dateutil, dateutils, asnhistory, lxml, ordered-set, mixbox, cybox, dnspython, dnspython3, ipasn-redis, ez-setup, future, passivetotal, olefile, pillow, pyeupi, pypssl, pytesseract, click, click-plugins, colorama, shodan, stix, tornado, url-normalize, urlarchiver, misp-modules
Successfully installed PyPDNS-1.3 argparse-1.4.0 asnhistory-2.0.4 certifi-2017.4.17 chardet-3.0.3 click-6.7 click-plugins-1.0.3 colorama-0.3.9 cybox-2.1.0.14 dateutils-0.6.6 dnspython-1.15.0 dnspython3-1.15.0 ez-setup-0.9 future-0.16.0 idna-2.5 ipasn-redis-2.0 lxml-3.8.0 misp-modules-1.0 mixbox-1.0.2 olefile-0.44 ordered-set-2.0.2 passivetotal-1.0.30 pillow-4.1.1 pyeupi-1.0 pypssl-2.1 pytesseract-0.1.7 python-dateutil-2.6.0 pytz-2017.2 redis-2.10.5 requests-2.17.3 requests-cache-0.4.13 shodan-1.6.8 six-1.10.0 stix-1.2.0.4 tornado-4.5.1 url-normalize-1.3.1 urlarchiver-0.2 urllib3-1.21.1

**Comment by Rafiot** [comment]
I cannot give it a try right now, so the debug might be tricky.

Does the error show up if you do `misp-modules -s` and/or `misp-modules`? (without `sudo -u www-data`)

**Comment by maisenhe** [comment]
The error only happens when I try to launch it with the -s . If I use just the misp-modules it launches but I have to do that in a screen so I can leave them up so it's not perfect. Thanks again

**Comment by se-te** [comment]
Hi!

On my MISP instances, this is due to the new version of shodan that has a syntax error in a function.

Please have a look at https://github.com/achillean/shodan-python/issues/39.

I have changed by hand the affected function in the vi /usr/local/lib/python3.5/dist-packages/shodan/helpers.py file on my system and... voilà! misp-modules start again.

**Comment by Rafiot** [comment]
@se-te can you do a PR?

**Comment by se-te** [comment]
Hi @Rafiot, not sure how I can do that. Isn't /usr/local/lib/python3.5/dist-packages/shodan/helpers.py non-MISP code?

According to https://github.com/achillean/shodan-python/issues/39 the issue they had with this is now closed. The new version of the Shodan library includes the fix.

I could not figure out exactly how the new library is related to misp-modules, but on a test machine today I cloned https://github.com/MISP/misp-modules.git again and pip3 re-installed everything. Good news: although the version number has not changed (still shodan-1.6.9), it seems that the function has been corrected and the misp-modules start well.
